freqtrade/exchange/exchange.py

```diff
@@ -1368,12 +1377,87 @@ async def _async_get_candle_history(self, pair: str, timeframe: str,
                                        f'for pair {pair}. Message: {e}') from e
 
+    # Directory name for a pair, e.g., given `ETH/BTC` returns `ETH-BTC`.
+    def _pair_dir(self, pair: str) -> str:
+        return pair.replace("/", "-")
+
+    # Returns the directory path that contains the intermediate trade files for a given pair.
+    def _intermediate_trades_dir_for_pair(self, datadir: Path, pair: str) -> str:
+        tmpdata_dir = os.path.join(datadir, "trades-intermediate-parts")
+        tmpdata_dir = os.path.join(tmpdata_dir, self._pair_dir(pair))
+        return tmpdata_dir
+
+    # Returns the intermediate trade file name for a given pair starting with `from_id` trade ID
+    def _intermediate_trades_file(self, datadir: Path, pair: str, from_id: str,
+                                  mkdir: bool = True) -> str:
+        tmpdata_file = self._intermediate_trades_dir_for_pair(datadir, pair)
+        tmpdata_file = os.path.join(tmpdata_file, str(int(from_id)//1000000))
+        if mkdir:
+            Path(tmpdata_file).mkdir(parents=True, exist_ok=True)
+        tmpdata_file = os.path.join(tmpdata_file, self._pair_dir(pair)+"_"+str(from_id)+".json")
+        return tmpdata_file
+
     # Fetch historic trades
+    @retrier_async
+    async def _async_fetch_trades_from_file(self, datafile: str) -> List[List]:
+        # Open a file: file
+        file = open(datafile, mode='r')
+
+        # read all lines at once
+        json_string = file.read()
+
+        # close the file
+        file.close()
+        trades_list = json.loads(json_string)
+        return trades_list
+
+    # Builds the interval tree (from the intermediate trade files) for a given pair
+    @retrier_async
+    async def _get_interval_tree_for_pair(self, datadir: Path, pair: str) -> IntervalTree:
+        cached_res = self._intermediate_data_cache.get(pair, None)
+        if cached_res:
+            return cached_res
+
+        logger.debug("Caching intervals for pair %s", pair)
+        cache_interval_tree = IntervalTree()
+        tmpdata_dir = self._intermediate_trades_dir_for_pair(datadir, pair)
+        if os.path.isdir(tmpdata_dir):
+            tmpdata_subdirs = sorted(os.listdir(tmpdata_dir))
+            for tmpdata_subdir in tmpdata_subdirs:
+                if tmpdata_subdir.isnumeric():
+                    tmpdata_subdir = os.path.join(tmpdata_dir, tmpdata_subdir)
+                    tmpdata_files = sorted(os.listdir(tmpdata_subdir))
+                    for tmpdata_file in tmpdata_files:
+                        if tmpdata_file.startswith(self._pair_dir(pair)):
+                            tmpdata_file = os.path.join(tmpdata_subdir, tmpdata_file)
+                            if os.path.isfile(tmpdata_file):
+                                trades_list = await self._async_fetch_trades_from_file(tmpdata_file)
+                                trades_list_from_id = trades_list[0][1]
+                                trades_list_to_id = trades_list[-1][1]
```

Comment:
I think this takes an assumption that will not hold depending on the exchange you're using this against.

While trade-id's for some exchanges are numeric and sequential - it's not the case for all exchanges (especially for exchanges that rely on time for iteration). 
As such, you can't assume that all id's between the first and last entry are present - as it could also be UUID's (which are completely random).